query-engine/connectors/mongodb-query-connector/src/value.rs

```diff
@@ -152,7 +152,7 @@ impl IntoBson for (&MongoDbType, PrismaValue) {
 
             // Double
             (MongoDbType::Double, PrismaValue::Int(i)) => Bson::Double(i as f64),
-            (MongoDbType::Double, PrismaValue::Float(f)) => Bson::Double(f.to_f64().convert(expl::MONGO_DOUBLE)?),
+            (MongoDbType::Double, PrismaValue::Float(f)) => bigdecimal_to_bson_double(f)?,
```

Comment:
Side-effect of `ScalarField::native_type` always returning a native type now. This is just a bug-fix that had not been caught before.